<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Downstream Analysis for Saturation Genome Editing Screens • sgeasy</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Downstream Analysis for Saturation Genome Editing Screens">
<meta name="description" content="Provides a user-friendly workflow for analyzing saturation genome editing (SGE) screen data. Includes functions for data normalization using control oligos, DESeq2-based differential abundance analysis, and visualization utilities for quality control and results exploration. Designed to work with QUANTS output and VEP annotations.">
<meta property="og:description" content="Provides a user-friendly workflow for analyzing saturation genome editing (SGE) screen data. Includes functions for data normalization using control oligos, DESeq2-based differential abundance analysis, and visualization utilities for quality control and results exploration. Designed to work with QUANTS output and VEP annotations.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">sgeasy</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/sge-analysis-workflow.html">SGE Downstream Analysis Workflow</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/team113sanger/sgeasy/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="sgeasy">sgeasy<a class="anchor" aria-label="anchor" href="#sgeasy"></a>
</h1></div>
<p>Downstream analysis tools for Saturation Genome Editing (SGE) screens. This package was developed by David Adams’ lab at WTSI and contains functions for integrating SGE data across screens to the gene-level and methods.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"t113sanger/sgeasy"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Or install locally</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>sgedown provides a workflow for analyzing SGE screen data:</p>
<ul>
<li>
<strong>Normalization</strong>: Size factor estimation using control oligos (synonymous/intronic variants)</li>
<li>
<strong>Differential Analysis</strong>: DESeq2-based differential abundance across time points</li>
<li>
<strong>Meta-analysis</strong>: Combining variant scores from across screens</li>
<li>
<strong>Visualization</strong>: Quality control plots and sample comparisons</li>
</ul>
</div>
<div class="section level2">
<h2 id="quick-start">Quick Start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/team113sanger/sgeasy">sgeasy</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create normalization matrix from neutral variants</span></span>
<span><span class="va">norm_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/create_normalization_matrix.html">create_normalization_matrix</a></span><span class="op">(</span><span class="va">annotated_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create count matrix for all variants</span></span>
<span><span class="va">count_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/create_count_matrix.html">create_count_matrix</a></span><span class="op">(</span><span class="va">annotated_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run differential abundance analysis</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/run_differential_analysis.html">run_differential_analysis</a></span><span class="op">(</span></span>
<span>  count_matrix <span class="op">=</span> <span class="va">count_matrix</span>,</span>
<span>  normalization_matrix <span class="op">=</span> <span class="va">norm_matrix</span>,</span>
<span>  sample_metadata <span class="op">=</span> <span class="va">metadata</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Access results</span></span>
<span><span class="va">deseq_results</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">results</span></span>
<span><span class="va">rlog_data</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">rlog</span></span>
<span><span class="va">summary_table</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">contrast_summary</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<p>See the package vignette for a complete workflow tutorial:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"sge-analysis-workflow"</span>, package <span class="op">=</span> <span class="st">"sgedown"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/team113sanger/sgeasy/">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing sgeasy</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Jamie Billington <br><small class="roles"> Author, maintainer </small>   </li>
<li>Andrew Waters <br><small class="roles"> Author </small>   </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jamie Billington, Andrew Waters.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
