<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ sgeasy</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">sgeasy</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/sge-analysis-workflow.html">SGE Downstream Analysis Workflow</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/team113sanger/sgeasy/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="workflow-facades">Workflow Facades<a class="anchor" aria-label="anchor" href="#workflow-facades"></a></h2>

      <div class="section-desc"><p>High-level functions for streamlined analysis</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="load_sge_data.html">load_sge_data()</a></code>

        </dt>
        <dd>Load SGE data from files</dd>

        <dt>

          <code><a href="prepare_count_matrices.html">prepare_count_matrices()</a></code>

        </dt>
        <dd>Prepare count matrices for analysis</dd>

        <dt>

          <code><a href="analyze_screens.html">analyze_screens()</a></code>

        </dt>
        <dd>Run differential analysis on all screens</dd>

        <dt>

          <code><a href="generate_qc_plots.html">generate_qc_plots()</a></code>

        </dt>
        <dd>Generate QC plots</dd>

        <dt>

          <code><a href="generate_results.html">generate_results()</a></code>

        </dt>
        <dd>Generate analysis results</dd>

        <dt>

          <code><a href="prepare_gene_level_data.html">prepare_gene_level_data()</a></code>

        </dt>
        <dd>Prepare gene-level dataset from targeton results</dd>
      </dl></div><div class="section level2">
      <h2 id="classes">Classes<a class="anchor" aria-label="anchor" href="#classes"></a></h2>

      <div class="section-desc"><p>S7 class definitions for SGE data structures</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="sgeasy-classes.html">sgeasy-classes</a></code>

        </dt>
        <dd>S7 Class Definitions for sgeasy</dd>

        <dt>

          <code><a href="SGEData.html">SGEData()</a></code>

        </dt>
        <dd>SGEData Class</dd>

        <dt>

          <code><a href="SGEResults.html">SGEResults()</a></code>

        </dt>
        <dd>SGEResults Class</dd>
      </dl></div><div class="section level2">
      <h2 id="normalization">Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a></h2>

      <div class="section-desc"><p>Size factor estimation from control data</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="estimate_size_factors.html">estimate_size_factors()</a></code>

        </dt>
        <dd>Estimate size factors from control data</dd>

        <dt>

          <code><a href="compute_control_size_factors.html">compute_control_size_factors()</a></code>

        </dt>
        <dd>Compute size factors from control oligos</dd>
      </dl></div><div class="section level2">
      <h2 id="data-wrangling">Data Wrangling<a class="anchor" aria-label="anchor" href="#data-wrangling"></a></h2>

      <div class="section-desc"><p>Matrix creation and filtering</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="create_normalization_matrix.html">create_normalization_matrix()</a></code>

        </dt>
        <dd>Create normalization matrix from neutral variants</dd>

        <dt>

          <code><a href="create_count_matrix.html">create_count_matrix()</a></code>

        </dt>
        <dd>Create count matrix from annotated data</dd>

        <dt>

          <code><a href="filter_by_counts.html">filter_by_counts()</a></code>

        </dt>
        <dd>Filter variants by count thresholds</dd>

        <dt>

          <code><a href="slim_consequence.html">slim_consequence()</a></code>

        </dt>
        <dd>Simplify variant consequence annotations</dd>

        <dt>

          <code><a href="calculate_position_effect.html">calculate_position_effect()</a></code>

        </dt>
        <dd>Calculate position effect ratios</dd>

        <dt>

          <code><a href="remove_artefacts.html">remove_artefacts()</a></code>

        </dt>
        <dd>Remove artefact variants</dd>
      </dl></div><div class="section level2">
      <h2 id="differential-analysis">Differential Analysis<a class="anchor" aria-label="anchor" href="#differential-analysis"></a></h2>

      <div class="section-desc"><p>DESeq2-based differential abundance and post-processing</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="run_differential_analysis.html">run_differential_analysis()</a></code>

        </dt>
        <dd>Run differential abundance analysis</dd>

        <dt>

          <code><a href="post_process.html">post_process()</a></code>

        </dt>
        <dd>Post-process screen results with annotation</dd>

        <dt>

          <code><a href="recalculate_screen_statistics.html">recalculate_screen_statistics()</a></code>

        </dt>
        <dd>Recalculate screen statistics</dd>

        <dt>

          <code><a href="calculate_median_scores.html">calculate_median_scores()</a></code>

        </dt>
        <dd>Calculate median scores for log2 fold changes</dd>

        <dt>

          <code><a href="adjust_all_contrasts.html">adjust_all_contrasts()</a></code>

        </dt>
        <dd>Adjust statistics for all contrasts</dd>

        <dt>

          <code><a href="classify_all_contrasts.html">classify_all_contrasts()</a></code>

        </dt>
        <dd>Classify variants for all contrasts</dd>

        <dt>

          <code><a href="classify_variants.html">classify_variants()</a></code>

        </dt>
        <dd>Classify variants by consequence type</dd>

        <dt>

          <code><a href="get_replicated_variants.html">get_replicated_variants()</a></code>

        </dt>
        <dd>Find replicated variants</dd>

        <dt>

          <code><a href="reweight_replicated_variants.html">reweight_replicated_variants()</a></code>

        </dt>
        <dd>Reweight replicated variants using inverse variance weighting</dd>
      </dl></div><div class="section level2">
      <h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a></h2>

      <div class="section-desc"><p>Plotting functions</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="plot_gene_level_waterfall.html">plot_gene_level_waterfall()</a></code>

        </dt>
        <dd>Plot functional scores along genomic coordinates</dd>

        <dt>

          <code><a href="plot_condition_scores.html">plot_condition_scores()</a></code>

        </dt>
        <dd>Plot z-scores for a single condition contrast</dd>

        <dt>

          <code><a href="plot_all_conditions.html">plot_all_conditions()</a></code>

        </dt>
        <dd>Generate plots for all condition contrasts</dd>

        <dt>

          <code><a href="write_waterfall_plots.html">write_waterfall_plots()</a></code>

        </dt>
        <dd>Write waterfall plots to files</dd>

        <dt>

          <code><a href="plot_ridgeline.html">plot_ridgeline()</a></code>

        </dt>
        <dd>Plot ridgeline density distributions by group</dd>

        <dt>

          <code><a href="plot_consequence_jitter.html">plot_consequence_jitter()</a></code>

        </dt>
        <dd>Plot jittered strip chart by group</dd>

        <dt>

          <code><a href="plot_sample_scatter.html">plot_sample_scatter()</a></code>

        </dt>
        <dd>Create sample comparison scatter plots</dd>

        <dt>

          <code><a href="sample_distance_matrix.html">sample_distance_matrix()</a></code>

        </dt>
        <dd>Create sample distance matrix heatmap</dd>

        <dt>

          <code><a href="plot_distance_matrices.html">plot_distance_matrices()</a></code>

        </dt>
        <dd>Save distance matrix plots to files</dd>

        <dt>

          <code><a href="build_sample_pca.html">build_sample_pca()</a></code>

        </dt>
        <dd>Build PCA plot from rlog data</dd>

        <dt>

          <code><a href="plot_screen_pcas.html">plot_screen_pcas()</a></code>

        </dt>
        <dd>Save PCA plots to files</dd>

        <dt>

          <code><a href="position_effect_plot.html">position_effect_plot()</a></code>

        </dt>
        <dd>Plot position effect ratios</dd>

        <dt>

          <code><a href="plot_position_effects.html">plot_position_effects()</a></code>

        </dt>
        <dd>Save position effect ratio plots to files</dd>

        <dt>

          <code><a href="consequence_colours.html">consequence_colours</a></code>

        </dt>
        <dd>Default color palette for consequence types</dd>

        <dt>

          <code><a href="classification_shapes.html">classification_shapes</a></code>

        </dt>
        <dd>Default shape palette for functional classifications</dd>

        <dt>

          <code><a href="default_consequences.html">default_consequences</a></code>

        </dt>
        <dd>Default consequence categories for distribution plots</dd>
      </dl></div><div class="section level2">
      <h2 id="domain-visualization">Domain Visualization<a class="anchor" aria-label="anchor" href="#domain-visualization"></a></h2>

      <div class="section-desc"><p>Protein domain fetching and heatmap visualization</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="fetch_domains_ted.html">fetch_domains_ted()</a></code>

        </dt>
        <dd>Fetch protein domains from TED (The Encyclopedia of Domains)</dd>

        <dt>

          <code><a href="fetch_domains_interpro.html">fetch_domains_interpro()</a></code>

        </dt>
        <dd>Fetch protein domains from InterPro REST API</dd>

        <dt>

          <code><a href="fetch_domains_ensembl.html">fetch_domains_ensembl()</a></code>

        </dt>
        <dd>Fetch protein domains from Ensembl REST API</dd>

        <dt>

          <code><a href="fetch_plddt_alphafold.html">fetch_plddt_alphafold()</a></code>

        </dt>
        <dd>Fetch per-residue pLDDT scores from AlphaFold Database</dd>

        <dt>

          <code><a href="merge_overlapping_domains.html">merge_overlapping_domains()</a></code>

        </dt>
        <dd>Merge overlapping domain ranges</dd>

        <dt>

          <code><a href="filter_protein_altering_variants.html">filter_protein_altering_variants()</a></code>

        </dt>
        <dd>Filter dataset to protein_altering variants for heatmap visualization</dd>

        <dt>

          <code><a href="prepare_amino_acid_heatmap_data.html">prepare_amino_acid_heatmap_data()</a></code>

        </dt>
        <dd>Prepare data for amino acid heatmap visualization</dd>

        <dt>

          <code><a href="plot_domain_track.html">plot_domain_track()</a></code>

        </dt>
        <dd>Plot protein domain track</dd>

        <dt>

          <code><a href="plot_plddt_track.html">plot_plddt_track()</a></code>

        </dt>
        <dd>Plot pLDDT confidence track</dd>

        <dt>

          <code><a href="plot_amino_acid_heatmap.html">plot_amino_acid_heatmap()</a></code>

        </dt>
        <dd>Plot amino acid substitution heatmap</dd>

        <dt>

          <code><a href="plot_domain_heatmap.html">plot_domain_heatmap()</a></code>

        </dt>
        <dd>Plot combined domain track and amino acid heatmap</dd>

        <dt>

          <code><a href="amino_acid_order.html">amino_acid_order</a></code>

        </dt>
        <dd>Standard amino acid ordering for heatmap y-axis</dd>

        <dt>

          <code><a href="classification_heatmap_colours.html">classification_heatmap_colours</a></code>

        </dt>
        <dd>Classification colours for amino acid heatmap</dd>

        <dt>

          <code><a href="plddt_colours.html">plddt_colours</a></code>

        </dt>
        <dd>AlphaFold pLDDT confidence colour scheme</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jamie Billington, Andrew Waters.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

