<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>SGE Downstream Analysis Workflow • sgeasy</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="SGE Downstream Analysis Workflow">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">sgeasy</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/sge-analysis-workflow.html">SGE Downstream Analysis Workflow</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/team113sanger/sgeasy/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>SGE Downstream Analysis Workflow</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/team113sanger/sgeasy/blob/master/vignettes/sge-analysis-workflow.Rmd"><code>vignettes/sge-analysis-workflow.Rmd</code></a></small>
      <div class="d-none name"><code>sge-analysis-workflow.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates a complete workflow for analyzing
Saturation Genome Editing (SGE) screen data using the sgeasy
package.</p>
<div class="section level3">
<h3 id="quick-start-with-stage-wrappers">Quick Start with Stage Wrappers<a class="anchor" aria-label="anchor" href="#quick-start-with-stage-wrappers"></a>
</h3>
<p>For a streamlined workflow, use the stage wrapper functions:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/team113sanger/sgeasy">sgeasy</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Stage 1: Load data (returns SGEData object)</span></span>
<span><span class="va">sge_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_sge_data.html">load_sge_data</a></span><span class="op">(</span></span>
<span>  count_files <span class="op">=</span> <span class="st">"inputs/"</span>,</span>
<span>  metadata_file <span class="op">=</span> <span class="st">"metadata/screen_metadata.tsv"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">sge_data</span></span>
<span><span class="co">#&gt; ── SGE Data ──</span></span>
<span><span class="co">#&gt; • Variants: 50,000</span></span>
<span><span class="co">#&gt; • Samples: 16</span></span>
<span><span class="co">#&gt; • Conditions: 4</span></span>
<span><span class="co">#&gt; • Annotation: loaded</span></span>
<span></span>
<span><span class="co"># Stage 2: Prepare matrices (accepts SGEData directly)</span></span>
<span><span class="va">matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_count_matrices.html">prepare_count_matrices</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">sge_data</span>,</span>
<span>  exclude_samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"bad_sample"</span><span class="op">)</span>,</span>
<span>  min_counts <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Stage 3: Run analysis (can pass SGEData for metadata)</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyze_screens.html">analyze_screens</a></span><span class="op">(</span></span>
<span>  complete_matrices <span class="op">=</span> <span class="va">matrices</span><span class="op">$</span><span class="va">complete_matrices</span>,</span>
<span>  normalisation_matrices <span class="op">=</span> <span class="va">matrices</span><span class="op">$</span><span class="va">normalisation_matrices</span>,</span>
<span>  metadata <span class="op">=</span> <span class="va">sge_data</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Stage 4: Generate QC plots</span></span>
<span><span class="fu"><a href="../reference/generate_qc_plots.html">generate_qc_plots</a></span><span class="op">(</span></span>
<span>  rlog_results <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">rlog_results</span>,</span>
<span>  output_dir <span class="op">=</span> <span class="st">"results/qc"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Stage 5: Generate results</span></span>
<span><span class="va">final</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_results.html">generate_results</a></span><span class="op">(</span></span>
<span>  contrast_tables <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">contrast_tables</span>,</span>
<span>  annotation <span class="op">=</span> <span class="va">sge_data</span><span class="op">@</span><span class="va">annotation</span>,</span>
<span>  output_dir <span class="op">=</span> <span class="st">"results"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="detailed-workflow">Detailed Workflow<a class="anchor" aria-label="anchor" href="#detailed-workflow"></a>
</h3>
<p>The following sections provide a more detailed explanation of each
step. The workflow includes:</p>
<ol style="list-style-type: decimal">
<li>Reading and preparing count data</li>
<li>Joining with sample metadata and variant annotations</li>
<li>Filtering and removing artefacts</li>
<li>Normalizing using control variants (synonymous/intronic)</li>
<li>Running differential abundance analysis with DESeq2</li>
<li>Quality control visualization (distance matrices, PCA)</li>
<li>Post-processing and statistical adjustment</li>
<li>Gene-level analysis</li>
<li>Reweighting replicated variants</li>
<li>Visualization of results</li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>Load the required packages:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/team113sanger/sgeasy">sgeasy</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://fs.r-lib.org" class="external-link">fs</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org" class="external-link">readr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://glue.tidyverse.org/" class="external-link">glue</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-import">Data Import<a class="anchor" aria-label="anchor" href="#data-import"></a>
</h2>
<div class="section level3">
<h3 id="reading-count-files">Reading Count Files<a class="anchor" aria-label="anchor" href="#reading-count-files"></a>
</h3>
<p>SGE count data typically comes from multiple quantification files.
Use <code><a href="https://fs.r-lib.org/reference/dir_ls.html" class="external-link">fs::dir_ls()</a></code> to find all count files:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input_files</span> <span class="op">&lt;-</span> <span class="fu">fs</span><span class="fu">::</span><span class="fu"><a href="https://fs.r-lib.org/reference/dir_ls.html" class="external-link">dir_ls</a></span><span class="op">(</span></span>
<span>  <span class="st">"inputs/run_51287"</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"any"</span>,</span>
<span>  recurse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  glob <span class="op">=</span> <span class="st">"*lib_counts.tsv.gz"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Optionally filter files (e.g., exclude certain samples)</span></span>
<span><span class="va">input_files</span> <span class="op">&lt;-</span> <span class="va">input_files</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="va">input_files</span>, pattern <span class="op">=</span> <span class="st">"pel"</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Add additional runs if needed</span></span>
<span><span class="va">added_files</span> <span class="op">&lt;-</span> <span class="fu">fs</span><span class="fu">::</span><span class="fu"><a href="https://fs.r-lib.org/reference/dir_ls.html" class="external-link">dir_ls</a></span><span class="op">(</span></span>
<span>  <span class="st">"inputs/run_47241"</span>,</span>
<span>  recurse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  glob <span class="op">=</span> <span class="st">"*lib_counts.tsv.gz"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">added_files</span> <span class="op">&lt;-</span> <span class="va">added_files</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="va">added_files</span>, pattern <span class="op">=</span> <span class="st">"mave-qc"</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">input_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">input_files</span>, <span class="va">added_files</span><span class="op">)</span></span></code></pre></div>
<p>Read all count files into a single dataset:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts_dataset</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="va">input_files</span>, comment <span class="op">=</span> <span class="st">"##"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="loading-metadata">Loading Metadata<a class="anchor" aria-label="anchor" href="#loading-metadata"></a>
</h3>
<p>Load the screen metadata containing sample information:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">screen_metadata</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="st">"metadata/screen_metadata.tsv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate duration from condition (e.g., "Day7" -&gt; 7 - 4 = 3)</span></span>
<span><span class="va">screen_metadata</span> <span class="op">&lt;-</span> <span class="va">screen_metadata</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>duration <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu">str_remove</span><span class="op">(</span><span class="va">condition</span>, pattern <span class="op">=</span> <span class="st">"Day"</span><span class="op">)</span><span class="op">)</span> <span class="op">-</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="data-preparation">Data Preparation<a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h2>
<div class="section level3">
<h3 id="joining-counts-with-metadata">Joining Counts with Metadata<a class="anchor" aria-label="anchor" href="#joining-counts-with-metadata"></a>
</h3>
<p>Combine count data with sample metadata:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">complete_dataset</span> <span class="op">&lt;-</span> <span class="va">counts_dataset</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">screen_metadata</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SAMPLE"</span> <span class="op">=</span> <span class="st">"sequencing_id"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="sample-exclusion">Sample Exclusion<a class="anchor" aria-label="anchor" href="#sample-exclusion"></a>
</h3>
<p>Remove any problematic samples:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exclude_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SAMPLE_A_10_pel0000"</span><span class="op">)</span>  <span class="co"># Replace with actual sample names</span></span>
<span></span>
<span><span class="va">complete_dataset</span> <span class="op">&lt;-</span> <span class="va">complete_dataset</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">supplier_name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">exclude_samples</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">SAMPLE</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">exclude_samples</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">targeton_id</span> <span class="op">!=</span> <span class="st">"FRTW"</span><span class="op">)</span>  <span class="co"># Exclude specific targetons if needed</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="splitting-by-targeton">Splitting by Targeton<a class="anchor" aria-label="anchor" href="#splitting-by-targeton"></a>
</h3>
<p>If your data contains multiple targetons, split into separate
datasets:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">targeton_matrices</span> <span class="op">&lt;-</span> <span class="va">complete_dataset</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">targeton_id</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_split.html" class="external-link">group_split</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Name each list element by targeton_id</span></span>
<span><span class="va">targeton_matrices</span> <span class="op">&lt;-</span> <span class="va">targeton_matrices</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://rlang.r-lib.org/reference/set_names.html" class="external-link">set_names</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_chr</a></span><span class="op">(</span><span class="va">targeton_matrices</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">targeton_id</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="creating-count-matrices">Creating Count Matrices<a class="anchor" aria-label="anchor" href="#creating-count-matrices"></a>
</h3>
<p>Pivot the data to wide format with samples as columns.
<strong>Important:</strong> Use <code>names_prefix = "count_"</code> to
prepend a prefix to sample column names:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count_matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>  <span class="va">targeton_matrices</span>,</span>
<span>  <span class="va">pivot_wider</span>,</span>
<span>  names_from <span class="op">=</span> <span class="st">"supplier_name"</span>,</span>
<span>  values_from <span class="op">=</span> <span class="st">"COUNT"</span>,</span>
<span>  names_prefix <span class="op">=</span> <span class="st">"count_"</span>,</span>
<span>  id_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NAME"</span>, <span class="st">"SEQUENCE"</span>, <span class="st">"LENGTH"</span>, <span class="st">"targeton_id"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Handle duplicate sequences by keeping the first occurrence:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">unique_matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>  <span class="va">count_matrices</span>,</span>
<span>  <span class="op">~</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">.x</span>, <span class="va">SEQUENCE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="filtering-by-count-threshold">Filtering by Count Threshold<a class="anchor" aria-label="anchor" href="#filtering-by-count-threshold"></a>
</h3>
<p>Use <code><a href="../reference/filter_by_counts.html">filter_by_counts()</a></code> to remove low-count variants:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtered_matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>  <span class="va">unique_matrices</span>,</span>
<span>  <span class="va">filter_by_counts</span>,</span>
<span>  min_counts <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="adding-variant-annotations">Adding Variant Annotations<a class="anchor" aria-label="anchor" href="#adding-variant-annotations"></a>
</h2>
<p>Load VEP annotations and join with count data:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get annotation file path from metadata</span></span>
<span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="va">screen_metadata</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">vep_anno</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">master_annotation</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="va">annotations</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join annotations</span></span>
<span><span class="va">annotated_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>  <span class="va">filtered_matrices</span>,</span>
<span>  <span class="op">~</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>    <span class="va">.x</span>,</span>
<span>    <span class="va">master_annotation</span>,</span>
<span>    by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SEQUENCE"</span> <span class="op">=</span> <span class="st">"Seq"</span>, <span class="st">"targeton_id"</span> <span class="op">=</span> <span class="st">"Targeton_ID"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="removing-artefacts">Removing Artefacts<a class="anchor" aria-label="anchor" href="#removing-artefacts"></a>
</h3>
<p>Remove variants that lack valid sgRNA identifiers (artefacts):</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotated_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">annotated_counts</span>, <span class="va">remove_artefacts</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="normalization">Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h2>
<div class="section level3">
<h3 id="creating-normalization-matrices">Creating Normalization Matrices<a class="anchor" aria-label="anchor" href="#creating-normalization-matrices"></a>
</h3>
<p>Extract neutral variants (synonymous and intronic) for
normalization:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normalisation_matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>  <span class="va">annotated_counts</span>,</span>
<span>  <span class="va">create_normalization_matrix</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>This uses variants with no expected functional effect as controls for
estimating library size factors.</p>
</div>
<div class="section level3">
<h3 id="creating-input-matrices">Creating Input Matrices<a class="anchor" aria-label="anchor" href="#creating-input-matrices"></a>
</h3>
<p>Create the full count matrices for analysis:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">complete_matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>  <span class="va">annotated_counts</span>,</span>
<span>  <span class="va">create_count_matrix</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="differential-abundance-analysis">Differential Abundance Analysis<a class="anchor" aria-label="anchor" href="#differential-abundance-analysis"></a>
</h2>
<p>Run the complete differential abundance pipeline:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deseq_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html" class="external-link">map2</a></span><span class="op">(</span></span>
<span>  <span class="va">complete_matrices</span>,</span>
<span>  <span class="va">normalisation_matrices</span>,</span>
<span>  <span class="va">run_differential_analysis</span>,</span>
<span>  sample_metadata <span class="op">=</span> <span class="va">screen_metadata</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Each element of <code>deseq_results</code> is an
<code>SGEResults</code> S7 object. You can print any result to see a
summary:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deseq_results</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; ── SGE Analysis Results ──</span></span>
<span><span class="co">#&gt; • Variants: 12,345</span></span>
<span><span class="co">#&gt; • Samples: 16</span></span>
<span><span class="co">#&gt; • Contrasts: 3</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; Access data with @results, @rlog, @contrast_summary</span></span></code></pre></div>
<p>Access properties using the <code>@</code> operator:</p>
<ul>
<li>
<code>result@results</code> - DESeq2 results table</li>
<li>
<code>result@rlog</code> - Regularized log-transformed data</li>
<li>
<code>result@contrast_summary</code> - Combined contrast
statistics</li>
<li>
<code>result@metadata</code> - Analysis parameters
(condition_levels, alpha, etc.)</li>
</ul>
<p>The <code><a href="../reference/run_differential_analysis.html">run_differential_analysis()</a></code> function:</p>
<ol style="list-style-type: decimal">
<li>Estimates size factors from control variants</li>
<li>Creates a DESeq2 dataset with the experimental design</li>
<li>Applies control-derived normalization</li>
<li>Runs Wald tests for all condition contrasts</li>
<li>Computes regularized log transformation</li>
<li>Generates z-scores for visualization</li>
<li>Runs continuous analysis for rate estimation</li>
</ol>
<div class="section level3">
<h3 id="combining-results">Combining Results<a class="anchor" aria-label="anchor" href="#combining-results"></a>
</h3>
<p>Combine contrast tables from all targetons using <code>map_dfr</code>
with <code>.id</code> to preserve targeton identity. Use <code>@</code>
to access the <code>contrast_summary</code> property from each
<code>SGEResults</code> object:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrast_tables</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map_dfr.html" class="external-link">map_dfr</a></span><span class="op">(</span></span>
<span>  <span class="va">deseq_results</span>,</span>
<span>  \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">@</span><span class="va">contrast_summary</span>,</span>
<span>  .id <span class="op">=</span> <span class="st">"Targeton_ID"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="quality-control-visualization">Quality Control Visualization<a class="anchor" aria-label="anchor" href="#quality-control-visualization"></a>
</h2>
<div class="section level3">
<h3 id="sample-distance-matrices">Sample Distance Matrices<a class="anchor" aria-label="anchor" href="#sample-distance-matrices"></a>
</h3>
<p>Visualize sample-to-sample distances to check for outliers and batch
effects:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract rlog data from each SGEResults object</span></span>
<span><span class="va">rlog_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">deseq_results</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">@</span><span class="va">rlog</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dist_matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">rlog_results</span>, <span class="va">sample_distance_matrix</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_distance_matrices.html">plot_distance_matrices</a></span><span class="op">(</span><span class="va">dist_matrices</span>, output_dir <span class="op">=</span> <span class="st">"results/qc"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pca-plots">PCA Plots<a class="anchor" aria-label="anchor" href="#pca-plots"></a>
</h3>
<p>Create PCA plots to visualize sample clustering:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">screen_pcas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">rlog_results</span>, <span class="va">build_sample_pca</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_screen_pcas.html">plot_screen_pcas</a></span><span class="op">(</span><span class="va">screen_pcas</span>, output_dir <span class="op">=</span> <span class="st">"results/qc"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="post-processing">Post-Processing<a class="anchor" aria-label="anchor" href="#post-processing"></a>
</h2>
<p>Apply post-processing to combine results with annotations,
recalculate statistics, and classify variants:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">recalculated_contrast_tables</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/post_process.html">post_process</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">contrast_tables</span>,</span>
<span>  annotation <span class="op">=</span> <span class="va">master_annotation</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The <code><a href="../reference/post_process.html">post_process()</a></code> function:</p>
<ol style="list-style-type: decimal">
<li>Joins with the annotation file by SEQUENCE and Targeton_ID</li>
<li>Calculates median log2FoldChange from neutral variants</li>
<li>Adjusts all contrasts by subtracting medians and computing
z-scores</li>
<li>Classifies variants as depleted/enriched/unchanged based on FDR</li>
<li>Simplifies consequence annotations</li>
</ol>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<div class="section level3">
<h3 id="waterfall-plots-by-targeton">Waterfall Plots by Targeton<a class="anchor" aria-label="anchor" href="#waterfall-plots-by-targeton"></a>
</h3>
<p>Generate waterfall plots for each targeton and condition:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">individual_screens</span> <span class="op">&lt;-</span> <span class="va">recalculated_contrast_tables</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Targeton_ID</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_split.html" class="external-link">group_split</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rlang.r-lib.org/reference/set_names.html" class="external-link">set_names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">recalculated_contrast_tables</span><span class="op">$</span><span class="va">Targeton_ID</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plot_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/imap.html" class="external-link">imap</a></span><span class="op">(</span></span>
<span>  <span class="va">individual_screens</span>,</span>
<span>  <span class="op">~</span> <span class="fu"><a href="../reference/plot_all_conditions.html">plot_all_conditions</a></span><span class="op">(</span><span class="va">.x</span>, <span class="va">.y</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/write_waterfall_plots.html">write_waterfall_plots</a></span><span class="op">(</span><span class="va">plot_sets</span>, output_dir <span class="op">=</span> <span class="st">"results/screen"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="gene-level-analysis">Gene-Level Analysis<a class="anchor" aria-label="anchor" href="#gene-level-analysis"></a>
</h2>
<p>For multi-exon genes, combine results across targetons:</p>
<div class="section level3">
<h3 id="adding-exon-annotations">Adding Exon Annotations<a class="anchor" aria-label="anchor" href="#adding-exon-annotations"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exon_annotations</span> <span class="op">&lt;-</span> <span class="va">recalculated_contrast_tables</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Targeton_ID</span>, <span class="va">EXON</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Targeton_ID</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Exons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu">str_remove</span><span class="op">(</span><span class="va">EXON</span>, pattern <span class="op">=</span> <span class="st">"/6"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">Exons</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Targeton_ID</span>, <span class="va">Exons</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Exons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">Exons</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"6"</span>, <span class="st">"5"</span>, <span class="st">"4"</span>, <span class="st">"3"</span>, <span class="st">"2"</span>, <span class="st">"1"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gene_level_dataset</span> <span class="op">&lt;-</span> <span class="va">recalculated_contrast_tables</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">exon_annotations</span>, by <span class="op">=</span> <span class="st">"Targeton_ID"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="gene-level-plot">Gene-Level Plot<a class="anchor" aria-label="anchor" href="#gene-level-plot"></a>
</h3>
<p>Create a gene-level functional score plot faceted by exon:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">score_col</span> <span class="op">&lt;-</span> <span class="st">"adj_lfc_continuous"</span></span>
<span><span class="va">class_col</span> <span class="op">&lt;-</span> <span class="st">"functional_classification_continuous"</span></span>
<span><span class="va">fdr_col</span> <span class="op">&lt;-</span> <span class="st">"FDR_continuous"</span></span>
<span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="st">"MYGENE"</span></span>
<span></span>
<span><span class="va">gene_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gene_level_dataset</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">vcf_pos</span>,</span>
<span>  y <span class="op">=</span> <span class="va">.data</span><span class="op">[[</span><span class="va">score_col</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  shape <span class="op">=</span> <span class="va">.data</span><span class="op">[[</span><span class="va">class_col</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  colour <span class="op">=</span> <span class="va">slim_consequence</span>,</span>
<span>  fill <span class="op">=</span> <span class="va">slim_consequence</span></span>
<span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="va">.data</span><span class="op">[[</span><span class="va">fdr_col</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="fl">0.01</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">consequence_colours</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">consequence_colours</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_shape_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">classification_shapes</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_alpha.html" class="external-link">scale_alpha_discrete</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FDR &gt; 0.01"</span>, <span class="st">"FDR &lt;= 0.01"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"GRCh38 coordinate"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Functional score (Adjusted LFC)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Functional score Plot:"</span>, <span class="va">gene_id</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                               colour <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>    legend.box <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>    legend.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Exons</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="st">"results/gene_level_plot.png"</span>, <span class="va">gene_level</span>,</span>
<span>       width <span class="op">=</span> <span class="fl">25</span>, height <span class="op">=</span> <span class="fl">6</span>, units <span class="op">=</span> <span class="st">"in"</span>, dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="reweighting-replicated-variants">Reweighting Replicated Variants<a class="anchor" aria-label="anchor" href="#reweighting-replicated-variants"></a>
</h2>
<p>Some variants may be represented by multiple oligos. Use inverse
variance weighting to combine these measurements:</p>
<div class="section level3">
<h3 id="preparing-results-for-reweighting">Preparing Results for Reweighting<a class="anchor" aria-label="anchor" href="#preparing-results-for-reweighting"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="va">gene_level_dataset</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Targeton_ID</span>, <span class="va">SEQUENCE</span>, <span class="va">HGVSc</span>, <span class="va">HGVSp</span>, <span class="va">pam_mut_sgrna_id</span>,</span>
<span>         <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">"continuous"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="finding-replicated-variants">Finding Replicated Variants<a class="anchor" aria-label="anchor" href="#finding-replicated-variants"></a>
</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">replicated_variants</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_replicated_variants.html">get_replicated_variants</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span><span class="va">replicated_variants</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="applying-inverse-variance-weighting">Applying Inverse Variance Weighting<a class="anchor" aria-label="anchor" href="#applying-inverse-variance-weighting"></a>
</h3>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reweighted_variants</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/reweight_replicated_variants.html">reweight_replicated_variants</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span></code></pre></div>
<p>The reweighting strategy:</p>
<ul>
<li>Non-PAM observations receive weight = 1/SE^2</li>
<li>PAM-impacted observations receive zero weight if non-PAM
alternatives exist</li>
<li>PAM-only variants use normal weighting (better than nothing)</li>
</ul>
</div>
<div class="section level3">
<h3 id="comparing-classifications">Comparing Classifications<a class="anchor" aria-label="anchor" href="#comparing-classifications"></a>
</h3>
<p>Check how reweighting affects classifications:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reweighted_variants</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">HGVSc</span>, <span class="va">HGVSp</span>, <span class="va">functional_classification</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">results</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HGVSc"</span>, <span class="st">"HGVSp"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">HGVSc</span>, <span class="va">HGVSp</span>, <span class="va">functional_classification</span>,</span>
<span>         <span class="va">functional_classification_continuous</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">functional_classification</span> <span class="op">!=</span> <span class="va">functional_classification_continuous</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="weighted-score-plot">Weighted Score Plot<a class="anchor" aria-label="anchor" href="#weighted-score-plot"></a>
</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">combined_dataset</span> <span class="op">&lt;-</span> <span class="va">reweighted_variants</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">gene_level_dataset</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HGVSc"</span>, <span class="st">"HGVSp"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">weighted_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">combined_dataset</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">vcf_pos</span>,</span>
<span>  y <span class="op">=</span> <span class="va">combined_lfc</span>,</span>
<span>  shape <span class="op">=</span> <span class="va">functional_classification</span>,</span>
<span>  colour <span class="op">=</span> <span class="va">slim_consequence</span>,</span>
<span>  fill <span class="op">=</span> <span class="va">slim_consequence</span></span>
<span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="va">FDR</span> <span class="op">&lt;=</span> <span class="fl">0.01</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">consequence_colours</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">consequence_colours</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_shape_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">classification_shapes</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_alpha.html" class="external-link">scale_alpha_discrete</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FDR &gt; 0.01"</span>, <span class="st">"FDR &lt;= 0.01"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"GRCh38 coordinate"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Functional score (Weighted LFC)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Functional score Plot:"</span>, <span class="va">gene_id</span>, <span class="st">"- Weighted LFC"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                               colour <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>    legend.box <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>    legend.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Exons</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="st">"results/gene_level_plot_reweight.png"</span>, <span class="va">weighted_scores</span>,</span>
<span>       width <span class="op">=</span> <span class="fl">25</span>, height <span class="op">=</span> <span class="fl">6</span>, units <span class="op">=</span> <span class="st">"in"</span>, dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="protein-domain-visualization">Protein Domain Visualization<a class="anchor" aria-label="anchor" href="#protein-domain-visualization"></a>
</h2>
<p>The sgeasy package includes functions for creating amino acid
substitution heatmaps with protein domain annotations. This
visualization shows functional effects across the protein sequence. ###
Fetching Protein Domains</p>
<p>Fetch domain annotations from Ensembl for your transcript:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch domains for your transcript</span></span>
<span><span class="va">transcript_id</span> <span class="op">&lt;-</span> <span class="st">"ENST00000355451"</span>  <span class="co"># Replace with your transcript</span></span>
<span><span class="va">domains</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fetch_domains_ensembl.html">fetch_domains_ensembl</a></span><span class="op">(</span><span class="va">transcript_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Merge overlapping domain ranges</span></span>
<span><span class="va">merged_domains</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_overlapping_domains.html">merge_overlapping_domains</a></span><span class="op">(</span><span class="va">domains</span><span class="op">)</span></span>
<span><span class="va">merged_domains</span></span></code></pre></div>
<div class="section level3">
<h3 id="preparing-protein-altering-variant-data">Preparing Protein altering Variant Data<a class="anchor" aria-label="anchor" href="#preparing-protein-altering-variant-data"></a>
</h3>
<p>Filter your dataset to Protein altering variants and prepare for
heatmap visualization:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter to protein altering variants</span></span>
<span><span class="va">protein_altering_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_protein_altering_variants.html">filter_protein_altering_variants</a></span><span class="op">(</span></span>
<span>  <span class="va">combined_dataset</span>,</span>
<span>  exclude_mutators <span class="op">=</span> <span class="st">"custom"</span>,</span>
<span>  exclude_consequences <span class="op">=</span> <span class="st">"synonymous"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Prepare data for heatmap (aggregates by amino acid position)</span></span>
<span><span class="va">plot_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_amino_acid_heatmap_data.html">prepare_amino_acid_heatmap_data</a></span><span class="op">(</span></span>
<span>  <span class="va">protein_altering_df</span>,</span>
<span>  score_col <span class="op">=</span> <span class="st">"adj_score_continuous"</span>,</span>
<span>  fdr_col <span class="op">=</span> <span class="st">"FDR_continuous"</span>,</span>
<span>  fdr_threshold <span class="op">=</span> <span class="fl">0.01</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">plot_df</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="creating-the-combined-visualization">Creating the Combined Visualization<a class="anchor" aria-label="anchor" href="#creating-the-combined-visualization"></a>
</h3>
<p>Create a combined plot with domain track above the amino acid
heatmap:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Option 1: All-in-one with automatic domain fetching</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_domain_heatmap.html">plot_domain_heatmap</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">protein_altering_df</span>,</span>
<span>  transcript_id <span class="op">=</span> <span class="st">"ENST00000355451"</span>,</span>
<span>  height_ratio <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">15</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Option 2: With pre-fetched domains</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_domain_heatmap.html">plot_domain_heatmap</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">protein_altering_df</span>,</span>
<span>  domain_df <span class="op">=</span> <span class="va">merged_domains</span>,</span>
<span>  height_ratio <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">15</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Save the plot</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="st">"results/amino_acid_domain_heatmap.png"</span>, <span class="va">p</span>,</span>
<span>       width <span class="op">=</span> <span class="fl">8</span>, height <span class="op">=</span> <span class="fl">7</span>, units <span class="op">=</span> <span class="st">"in"</span>, dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="creating-individual-plot-components">Creating Individual Plot Components<a class="anchor" aria-label="anchor" href="#creating-individual-plot-components"></a>
</h3>
<p>For more control, create the domain track and heatmap separately:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get shared x-axis range</span></span>
<span><span class="va">x_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">plot_df</span><span class="op">$</span><span class="va">Protein_position</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create domain track</span></span>
<span><span class="va">domain_track</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_domain_track.html">plot_domain_track</a></span><span class="op">(</span></span>
<span>  <span class="va">merged_domains</span>,</span>
<span>  x_range <span class="op">=</span> <span class="va">x_range</span>,</span>
<span>  show_legend <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create amino acid heatmap</span></span>
<span><span class="va">amino_acid_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_amino_acid_heatmap.html">plot_amino_acid_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">plot_df</span>,</span>
<span>  x_range <span class="op">=</span> <span class="va">x_range</span>,</span>
<span>  show_x_axis <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine with patchwork</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="va">combined_plot</span> <span class="op">&lt;-</span> <span class="va">domain_track</span> <span class="op">/</span> <span class="va">amino_acid_plot</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">15</span><span class="op">)</span>, axes <span class="op">=</span> <span class="st">"collect_x"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Save</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="st">"results/custom_domain_heatmap.png"</span>, <span class="va">combined_plot</span>,</span>
<span>       width <span class="op">=</span> <span class="fl">8</span>, height <span class="op">=</span> <span class="fl">6</span>, units <span class="op">=</span> <span class="st">"in"</span>, dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="customizing-colors">Customizing Colors<a class="anchor" aria-label="anchor" href="#customizing-colors"></a>
</h3>
<p>You can customize the classification colors:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Custom classification colors</span></span>
<span><span class="va">my_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"unchanged"</span> <span class="op">=</span> <span class="st">"grey80"</span>,</span>
<span>  <span class="st">"enriched"</span>  <span class="op">=</span> <span class="st">"firebrick"</span>,</span>
<span>  <span class="st">"depleted"</span>  <span class="op">=</span> <span class="st">"steelblue"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_domain_heatmap.html">plot_domain_heatmap</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">protein_altering_df</span>,</span>
<span>  domain_df <span class="op">=</span> <span class="va">merged_domains</span>,</span>
<span>  classification_colors <span class="op">=</span> <span class="va">my_colors</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="filtering-domains-by-source">Filtering Domains by Source<a class="anchor" aria-label="anchor" href="#filtering-domains-by-source"></a>
</h3>
<p>Optionally filter to specific domain sources:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch only Pfam and SMART domains</span></span>
<span><span class="va">domains</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fetch_domains_ensembl.html">fetch_domains_ensembl</a></span><span class="op">(</span></span>
<span>  <span class="st">"ENST00000355451"</span>,</span>
<span>  domain_sources <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pfam"</span>, <span class="st">"SMART"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="exporting-results">Exporting Results<a class="anchor" aria-label="anchor" href="#exporting-results"></a>
</h2>
<p>Save the final results:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Gene-level weighted scores</span></span>
<span><span class="va">combined_dataset</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">HGVSc</span>, <span class="va">HGVSp</span>, <span class="va">combined_lfc</span>, <span class="va">functional_classification</span>, <span class="va">FDR</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">results</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HGVSc"</span>, <span class="st">"HGVSp"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html" class="external-link">write_tsv</a></span><span class="op">(</span><span class="st">"results/gene_level_weighted_scores.tsv"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jamie Billington, Andrew Waters.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
